build/builder.go

```diff
@@ -618,20 +617,13 @@ func (b *Builder) Add(doc zoekt.Document) error {
 	}
 
 	allowLargeFile := b.opts.IgnoreSizeMax(doc.Name)
-
-	// Adjust trigramMax for allowed large files so we don't exclude them.
```

Comment:
I maintained these semantics, but they were surprising to me -- why don't we enforce the max trigram limit for allowed large files?

Reply:
when someone allow lists a file we don't ever want to exclude it. We can revisit this, but we need to avoid surprising behaviour here. In particular people often allow list package manager lock files which contain a large number of shasums (and therefore a lot of trigrams)